src/System.Management.Automation/namespaces/FunctionProvider.cs

```diff
@@ -309,7 +309,7 @@ internal override object GetValueOfItem(object item)
         /// </returns>
         internal override IDictionary GetSessionStateTable()
         {
-            return SessionState.Internal.GetFunctionTable();
+            return (IDictionary)SessionState.Internal.GetFunctionTable();
```

Comment:
It is internal. Can we replace with IDictionary<string, FunctionInfo> too?

Reply:
No, it's generic for all providers

Reply:
I see this already was done in AliasProvider, VariableProvider and EnvironmentProvider.
(I search by `IDictionary GetSessionStateTable()`.)
So I believe we can benefit from this in all code base.

Reply:
@iSazonov I'm not sure I follow you here - are you suggesting that the signature of 

```csharp
class SessionStateProviderBase{
internal abstract IDictionary GetSessionStateTable();
}
```

should change?

Reply:
I don't understand how we can. IDictionary<T,U> does not have contravariant type parameters.

Reply:
Yes, it is internal.

Reply:
This should works:
`internal abstract IDictionary<string, CommandInfo> GetSessionStateTable();`

Reply:
> This should works:
> `internal abstract IDictionary<string, CommandInfo> GetSessionStateTable();`

EnvironmentProvider does not return a CommandInfo.

And the type system doesn't allow it either.
If that was allowed, you could get a dictionary of AliasInfo and insert a FunctionInfo.

The type parameter of the dictionary value is not contravariant, and hence it does not work.
If it was contravariant, the definition would look like 
```csharp
interface IDictionary<TKey, in TValue>{
} 
```
where `in` indicates that `TValue` is contravariant, i.e. that it enables you to use a more generic (less derived) type than originally specified.

But it isn't, for good reasons.

Reply:
Here is a good overview of [co/contra-variance](https://docs.microsoft.com/en-us/dotnet/standard/generics/covariance-and-contravariance)